internal/provider/namespace_export_sink_resource.go

```diff
@@ -395,13 +424,28 @@ func getSinkSpecFromModel(ctx context.Context, plan *namespaceExportSinkResource
 			return nil, diags
 		}
 
+		saId := gcsSpec.SaId.ValueString()
+		gcpProjectId := gcsSpec.GcpProjectId.ValueString()
+		if saId == "" && gcpProjectId == "" && gcsSpec.ServiceAccountEmail.ValueString() != "" {
+			saId, gcpProjectId = parseSAPrincipal(gcsSpec.ServiceAccountEmail.ValueString())
+
+		}
+
+		if saId == "" && gcpProjectId == "" {
```

Comment:
Should this be an OR? I think we need both of these pieces of information to successfully create the resource, right?